### ISSUE TITLE
Receving java.lang.IllegalArgumentException: access field inside provided object cannot be evaluated by the operator type: CONTAINS

### ISSUE THREAD

**Issue by MSKoWa** [aws-amplify/amplify-flutter#1931]
### Description

Hello,

I am sometimes receiving the following error which seems to be causes by one of my queries (observing). I have a list of Strings in my model and query certain elements using ".contains".  One line of the error states that: "java.util.Collections$UnmodifiableRandomAccessList cannot be cast to java.lang.String". I am wondering how there can be this "UnmodifiableRandomAccessList" although I just have a list of Strings. This is the error:

```
Reading subscription events has failed.
W/amplify:aws-datastore(31270): java.lang.IllegalArgumentException: access field inside provided object cannot be evaluated by the operator type: CONTAINS
W/amplify:aws-datastore(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.evaluate(QueryPredicateOperation.java:141)
W/amplify:aws-datastore(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateGroup.evaluate(QueryPredicateGroup.java:127)
W/amplify:aws-datastore(31270): 	at com.amplifyframework.datastore.syncengine.SubscriptionProcessor.lambda$subscriptionObservable$8$SubscriptionProcessor(SubscriptionProcessor.java:237)
W/amplify:aws-datastore(31270): 	at com.amplifyframework.datastore.syncengine.-$$Lambda$SubscriptionProcessor$YOw2A6cN9GaUbIKehklooffucIg.test(Unknown Source:6)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:46)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:63)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
W/amplify:aws-datastore(31270): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/amplify:aws-datastore(31270): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
W/amplify:aws-datastore(31270): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/amplify:aws-datastore(31270): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/amplify:aws-datastore(31270): 	at java.lang.Thread.run(Thread.java:919)
W/amplify:aws-datastore(31270): Caused by: java.lang.ClassCastException: java.util.Collections$UnmodifiableRandomAccessList cannot be cast to java.lang.String
W/amplify:aws-datastore(31270): 	at com.amplifyframework.core.model.query.predicate.ContainsQueryOperator.evaluate(ContainsQueryOperator.java:23)
W/amplify:aws-datastore(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.evaluate(QueryPredicateOperation.java:137)
W/amplify:aws-datastore(31270): 	... 14 more
W/amplify:aws-datastore(31270): Reading subscriptions buffer has failed.
W/amplify:aws-datastore(31270): java.lang.IllegalArgumentException: access field inside provided object cannot be evaluated by the operator type: CONTAINS
W/amplify:aws-datastore(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.evaluate(QueryPredicateOperation.java:141)
W/amplify:aws-datastore(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateGroup.evaluate(QueryPredicateGroup.java:127)
W/amplify:aws-datastore(31270): 	at com.amplifyframework.datastore.syncengine.SubscriptionProcessor.lambda$subscriptionObservable$8$SubscriptionProcessor(SubscriptionProcessor.java:237)
W/amplify:aws-datastore(31270): 	at com.amplifyframework.datastore.syncengine.-$$Lambda$SubscriptionProcessor$YOw2A6cN9GaUbIKehklooffucIg.test(Unknown Source:6)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:46)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:63)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
W/amplify:aws-datastore(31270): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
W/amplify:aws-datastore(31270): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/amplify:aws-datastore(31270): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
W/amplify:aws-datastore(31270): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/amplify:aws-datastore(31270): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/amplify:aws-datastore(31270): 	at java.lang.Thread.run(Thread.java:919)
W/amplify:aws-datastore(31270): Caused by: java.lang.ClassCastException: java.util.Collections$UnmodifiableRandomAccessList cannot be cast to java.lang.String
W/amplify:aws-datastore(31270): 	at com.amplifyframework.core.model.query.predicate.ContainsQueryOperator.evaluate(ContainsQueryOperator.java:23)
W/amplify:aws-datastore(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.evaluate(QueryPredicateOperation.java:137)
W/amplify:aws-datastore(31270): 	... 14 more
W/System.err(31270): io.reactivex.rxjava3.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.IllegalArgumentException: access field inside provided object cannot be evaluated by the operator type: CONTAINS
W/System.err(31270): 	at io.reactivex.rxjava3.internal.observers.EmptyCompletableObserver.onError(EmptyCompletableObserver.java:50)
2
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onError(CompletablePeek.java:95)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:156)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable$FlatMapCompletableMainObserver.onError(ObservableFlatMapCompletableCompletable.java:125)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.observers.DisposableLambdaObserver.onError(DisposableLambdaObserver.java:65)
W/System.err(31270): 	at io.reactivex.rxjava3.subjects.ReplaySubject$UnboundedReplayBuffer.replay(ReplaySubject.java:782)
W/System.err(31270): 	at io.reactivex.rxjava3.subjects.ReplaySubject.onError(ReplaySubject.java:385)
W/System.err(31270): 	at com.amplifyframework.datastore.syncengine.-$$Lambda$baDcOAPYRoCcJg8VmByDb2fJo9E.accept(Unknown Source:4)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
2
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onError(ObservableDoOnEach.java:117)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.observers.DisposableLambdaObserver.onError(DisposableLambdaObserver.java:65)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:281)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
W/System.err(31270): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/System.err(31270): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
W/System.err(31270): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/System.err(31270): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/System.err(31270): 	at java.lang.Thread.run(Thread.java:919)
W/System.err(31270): Caused by: java.lang.IllegalArgumentException: access field inside provided object cannot be evaluated by the operator type: CONTAINS
W/System.err(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.evaluate(QueryPredicateOperation.java:141)
W/System.err(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateGroup.evaluate(QueryPredicateGroup.java:127)
W/System.err(31270): 	at com.amplifyframework.datastore.syncengine.SubscriptionProcessor.lambda$subscriptionObservable$8$SubscriptionProcessor(SubscriptionProcessor.java:237)
W/System.err(31270): 	at com.amplifyframework.datastore.syncengine.-$$Lambda$SubscriptionProcessor$YOw2A6cN9GaUbIKehklooffucIg.test(Unknown Source:6)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:46)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:63)
W/System.err(31270): 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
W/System.err(31270): 	... 8 more
W/System.err(31270): Caused by: java.lang.ClassCastException: java.util.Collections$UnmodifiableRandomAccessList cannot be cast to java.lang.String
W/System.err(31270): 	at com.amplifyframework.core.model.query.predicate.ContainsQueryOperator.evaluate(ContainsQueryOperator.java:23)
W/System.err(31270): 	at com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.evaluate(QueryPredicateOperation.java:137)
W/System.err(31270): 	... 14 more
I/Process (31270): Sending signal. PID: 31270 SIG: 9
Lost connection to device.
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [X] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Storage

### Platforms

- [ ] iOS
- [X] Android

### Android Device/Emulator API Level

API 30, API 31

### Environment

```bash
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] VS Code (version 1.68.1)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!
```

### Dependencies

```bash
Dart SDK 2.17.6
Flutter SDK 3.0.5
app_version 4.0.1+147

dependencies:
- amplify_api 0.5.1 [amplify_api_plugin_interface amplify_core amplify_api_android amplify_api_ios collection flutter meta plugin_platform_interface]
- amplify_auth_cognito 0.5.1 [amplify_auth_cognito_android amplify_auth_cognito_ios amplify_auth_plugin_interface amplify_core collection flutter plugin_platform_interface]
- amplify_datastore 0.5.1 [flutter amplify_datastore_plugin_interface amplify_core plugin_platform_interface meta collection async]
- amplify_flutter 0.5.1 [amplify_analytics_plugin_interface amplify_api_plugin_interface amplify_auth_plugin_interface amplify_core amplify_datastore_plugin_interface amplify_flutter_android amplify_flutter_ios amplify_storage_plugin_interface collection flutter json_annotation meta plugin_platform_interface]
- app_settings 4.1.8 [flutter]
- auto_size_text 3.0.0 [flutter]
- bloc 8.0.3 [meta]
- charcode 1.3.1
- colorful_safe_area 0.3.0 [flutter]
- connectivity_plus 2.3.6 [flutter connectivity_plus_platform_interface connectivity_plus_linux connectivity_plus_macos connectivity_plus_web connectivity_plus_windows]
- cupertino_icons 1.0.5
- dartz 0.10.1
- division 0.9.0 [flutter]
- drift 1.7.0 [async convert collection js meta stream_channel sqlite3]
- easy_localization 3.0.1 [flutter shared_preferences intl args path easy_logger flutter_localizations]
- easy_localization_loader 1.0.0 [http csv yaml xml flutter]
- equatable 2.0.3 [collection meta]
- file_picker 4.6.1 [flutter flutter_web_plugins flutter_plugin_android_lifecycle plugin_platform_interface ffi path win32]
- fl_chart 0.36.2 [flutter equatable pedantic image]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_bloc 8.0.1 [flutter bloc provider]
- flutter_datetime_picker 1.5.1 [flutter]
- flutter_hooks 0.18.5 [flutter]
- flutter_keyboard_visibility 5.3.0 [meta flutter_keyboard_visibility_platform_interface flutter_keyboard_visibility_web flutter]
- flutter_local_notifications 9.7.0 [clock flutter flutter_local_notifications_linux flutter_local_notifications_platform_interface timezone]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_native_splash 2.2.0+1 [args flutter flutter_web_plugins image js lint meta path universal_io xml yaml]
- flutter_slidable 0.6.0 [flutter]
- flutter_time_picker_spinner 2.0.0 [flutter]
- freezed 2.0.3+1 [analyzer build build_config collection meta source_gen freezed_annotation json_annotation]
- gallery_saver 2.3.2 [flutter path_provider http path]
- get_it 7.2.0 [async collection]
- hydrated_bloc 8.1.0 [bloc hive meta synchronized]
- image_picker 0.8.5+3 [flutter image_picker_android image_picker_for_web image_picker_ios image_picker_platform_interface]
- intl 0.17.0 [clock path]
- json_annotation 4.5.0 [meta]
- keyboard_actions 4.0.1 [flutter]
- liquid_progress_indicator 0.4.0 [flutter]
- package_info_plus 1.4.2 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- path 1.8.1
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- permission_handler 10.0.0 [flutter meta permission_handler_android permission_handler_apple permission_handler_windows permission_handler_platform_interface]
- provider 6.0.3 [collection flutter nested]
- purchases_flutter 3.8.0 [flutter collection]
- rxdart 0.27.4
- splashscreen 1.3.5 [flutter]
- sqlite3_flutter_libs 0.5.8 [flutter]
- streaming_shared_preferences 2.0.0 [flutter meta shared_preferences]
- syncfusion_flutter_sliders 20.1.61 [flutter intl syncfusion_flutter_core]
- url_launcher 6.1.3 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- uuid 3.0.6 [crypto]

transitive dependencies:
- _fe_analyzer_shared 40.0.0 [meta]
- amplify_analytics_plugin_interface 0.5.1 [amplify_core flutter meta]
- amplify_api_android 0.5.1 [flutter]
- amplify_api_ios 0.5.1 [amplify_core flutter]
- amplify_api_plugin_interface 0.5.1 [amplify_core collection flutter json_annotation meta]
- amplify_auth_cognito_android 0.5.1 [flutter]
- amplify_auth_cognito_ios 0.5.1 [amplify_core flutter]
- amplify_auth_plugin_interface 0.5.1 [amplify_core flutter meta]
- amplify_core 0.5.1 [collection date_time_format flutter meta plugin_platform_interface uuid]
- amplify_datastore_plugin_interface 0.5.1 [flutter meta collection amplify_core]
- amplify_flutter_android 0.5.1 [flutter]
- amplify_flutter_ios 0.5.1 [amplify_core flutter]
- amplify_storage_plugin_interface 0.5.1 [flutter meta amplify_core]
- analyzer 4.1.0 [_fe_analyzer_shared collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- archive 3.3.0 [crypto path]
- args 2.3.1
- async 2.8.2 [collection meta]
- build 2.3.0 [analyzer async convert crypto glob logging meta path]
- build_config 1.0.0 [checked_yaml json_annotation path pubspec_parse yaml]
- characters 1.2.0
- checked_yaml 2.0.1 [json_annotation source_span yaml]
- clock 1.1.0
- collection 1.16.0
- connectivity_plus_linux 1.3.1 [flutter connectivity_plus_platform_interface meta nm]
- connectivity_plus_macos 1.2.4 [connectivity_plus_platform_interface flutter]
- connectivity_plus_platform_interface 1.2.1 [flutter meta plugin_platform_interface]
- connectivity_plus_web 1.2.3 [connectivity_plus_platform_interface flutter_web_plugins flutter]
- connectivity_plus_windows 1.2.2 [connectivity_plus_platform_interface flutter]
- convert 3.0.2 [typed_data]
- cross_file 0.3.3+1 [js meta]
- crypto 3.0.2 [typed_data]
- csv 5.0.1
- dart_style 2.2.3 [analyzer args path pub_semver source_span]
- date_time_format 2.0.1
- dbus 0.7.3 [args ffi meta xml]
- easy_logger 0.0.2 [flutter]
- ffi 1.2.1
- file 6.1.2 [meta path]
- flutter_keyboard_visibility_platform_interface 2.0.0 [flutter meta plugin_platform_interface]
- flutter_keyboard_visibility_web 2.0.0 [flutter_keyboard_visibility_platform_interface flutter_web_plugins flutter]
- flutter_local_notifications_linux 0.5.0+1 [flutter flutter_local_notifications_platform_interface dbus path xdg_directories]
- flutter_local_notifications_platform_interface 5.0.0 [flutter plugin_platform_interface]
- flutter_plugin_android_lifecycle 2.0.6 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- freezed_annotation 2.0.3 [collection json_annotation meta]
- glob 2.1.0 [async collection file path string_scanner]
- hive 2.2.3 [meta crypto]
- http 0.13.4 [async http_parser meta path]
- http_parser 4.0.1 [collection source_span string_scanner typed_data]
- image 3.1.3 [archive meta xml]
- image_picker_android 0.8.5+1 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface]
- image_picker_for_web 2.1.8 [flutter flutter_web_plugins image_picker_platform_interface]
- image_picker_ios 0.8.5+5 [flutter image_picker_platform_interface]
- image_picker_platform_interface 2.5.0 [cross_file flutter http plugin_platform_interface]
- js 0.6.4
- lint 1.8.2
- logging 1.0.2
- material_color_utilities 0.1.4
- meta 1.7.0
- nested 1.0.0 [flutter]
- nm 0.5.0 [dbus]
- package_config 2.1.0 [path]
- package_info_plus_linux 1.0.5 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.3.0 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.5 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 1.0.5 [package_info_plus_platform_interface ffi flutter win32]
- path_provider_android 2.0.15 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.10 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.4 [flutter platform plugin_platform_interface]
- path_provider_windows 2.0.7 [ffi flutter path path_provider_platform_interface win32]
- pedantic 1.11.1
- permission_handler_android 10.0.0 [flutter permission_handler_platform_interface]
- permission_handler_apple 9.0.4 [flutter permission_handler_platform_interface]
- permission_handler_platform_interface 3.7.0 [flutter meta plugin_platform_interface]
- permission_handler_windows 0.1.0 [flutter permission_handler_platform_interface]
- petitparser 5.0.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- process 4.2.4 [file path platform]
- pub_semver 2.1.1 [collection meta]
- pubspec_parse 1.2.0 [checked_yaml collection json_annotation pub_semver yaml]
- shared_preferences 2.0.15 [flutter shared_preferences_android shared_preferences_ios shared_preferences_linux shared_preferences_macos shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- shared_preferences_android 2.0.12 [flutter shared_preferences_platform_interface]
- shared_preferences_ios 2.1.1 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.1.1 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_macos 2.0.4 [flutter shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.0.0 [flutter]
- shared_preferences_web 2.0.4 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.1.1 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sky_engine 0.0.99
- source_gen 1.2.2 [analyzer async build dart_style glob meta path source_span yaml]
- source_span 1.8.2 [collection path term_glyph]
- sqlite3 1.7.2 [collection ffi js meta path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- syncfusion_flutter_core 20.1.61 [vector_math flutter]
- synchronized 3.0.0+2
- term_glyph 1.2.0
- timezone 0.8.0 [path]
- typed_data 1.3.1 [collection]
- universal_io 2.0.4 [collection crypto meta typed_data]
- url_launcher_android 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.0 [flutter plugin_platform_interface]
- url_launcher_web 2.0.12 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- vector_math 2.1.2
- watcher 1.0.1 [async path]
- win32 2.6.1 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- xml 5.4.1 [collection meta petitparser]
- yaml 3.1.1 [collection source_span string_scanner]
```

### Device

Xiao Mi Pocophone F1

### OS

Android 10

### CLI Version

9.1.0

**Comment by HuiSF** [comment]
Hi @MSKoWa sorry for the delay, somehow this issue fell off my radar.

Can you provide your code example invoking `observe` API with the `.contains` predicate?

**Comment by MSKoWa** [comment]
Oh sorry, I already resolved this one and forget to close it.

The reason why I got this error was not caused by Amplify. I am using the flutter freezed library that resolves a list in the class definition to a unmodifiable list when it comes to accessing the class field. This is why I got the error, I tried to modify the unmodifiable list instead of using the copy with functionality which is supposed to be used instead
